clients/src/main/java/org/apache/kafka/clients/ClientUtils.java

```diff
@@ -108,10 +108,15 @@ public static ChannelBuilder createChannelBuilder(AbstractConfig config, Time ti
 
     static List<InetAddress> resolve(String host, ClientDnsLookup clientDnsLookup) throws UnknownHostException {
         InetAddress[] addresses = InetAddress.getAllByName(host);
-        if (ClientDnsLookup.USE_ALL_DNS_IPS == clientDnsLookup) {
-            return filterPreferredAddresses(addresses);
-        } else {
-            return Collections.singletonList(addresses[0]);
+        switch (clientDnsLookup) {
+            case DEFAULT:
+                log.warn("Configuration 'client.dns.lookup' value 'default' is deprecated and will be removed in the future version.");
```

Comment:
This will warn on every resolve. We should probably add the warning when we get the config value from `ConsumerConfig`, `ProducerConfig` and `AdminClientConfig`.

Reply:
ok. will move the warn there.

Reply:
I'm adding the check and the warning to the constructor of KafkaProducer, KafkaConsumer and KafkaAdminClient because the log context does not exist in the Config classes.

Reply:
Nevermind. I've added the warning to Config classes.